backend/app/api/endpoints/adapter/subscriptions.py

```
@@ -90,6 +92,36 @@ def create_subscription(
 
+# ========== Discover Endpoint ==========
+# NOTE: This static route MUST be defined before /{subscription_id} dynamic routes
+
+
+@router.get("/discover", response_model=DiscoverSubscriptionsListResponse)
+def discover_subscriptions(
+    sort_by: str = Query("popularity", description="Sort by: 'popularity' or 'recent'"),
+    search: Optional[str] = Query(None, description="Search query"),
+    page: int = Query(1, ge=1, description="Page number"),
+    limit: int = Query(50, ge=1, le=100, description="Items per page"),
+    db: Session = Depends(get_db),
```

Ruff (0.14.13)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)